docs/source/tutorials/improving_ml_performance.ipynb

```diff
@@ -0,0 +1,3815 @@
+       "706  [714, 707, 709, 710, 711, 712, 713, 708, 715]   \n",
+       "707  [714, 706, 709, 710, 711, 712, 713, 708, 715]   \n",
+       "708  [714, 707, 706, 709, 710, 711, 712, 713, 715]   \n",
+       "709  [714, 707, 706, 710, 711, 712, 713, 708, 715]   \n",
+       "710  [714, 707, 706, 709, 711, 712, 713, 708, 715]   \n",
+       "711  [714, 707, 706, 709, 710, 712, 713, 708, 715]   \n",
+       "712  [714, 707, 706, 709, 710, 711, 713, 708, 715]   \n",
+       "713  [714, 707, 706, 709, 710, 711, 712, 708, 715]   \n",
+       "714  [707, 706, 709, 710, 711, 712, 713, 708, 715]   \n",
+       "715  [714, 707, 706, 709, 710, 711, 712, 713, 708]   \n",
+       "\n",
+       "     distance_to_nearest_neighbor  \n",
+       "2                             0.0  \n",
+       "260                           0.0  \n",
+       "384                           0.0  \n",
+       "463                           0.0  \n",
+       "540                           0.0  \n",
+       "592                           0.0  \n",
+       "706                           0.0  \n",
+       "707                           0.0  \n",
+       "708                           0.0  \n",
+       "709                           0.0  \n",
+       "710                           0.0  \n",
+       "711                           0.0  \n",
+       "712                           0.0  \n",
+       "713                           0.0  \n",
+       "714                           0.0  \n",
+       "715                           0.0  "
+      ]
+     },
+     "execution_count": 15,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "exact_duplicates = full_duplicate_results[(full_duplicate_results[\"is_near_duplicate_issue\"] == True) & (full_duplicate_results[\"near_duplicate_score\"] == 0.0)].sort_values(\"near_duplicate_score\")\n",
+    "exact_duplicates"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9672b692-fc3e-4467-b5c4-4c367b55eb30",
+   "metadata": {},
+   "source": [
+    "To remove the exact duplicates that occur between our training and test sets from our training data, let's define the last index of data points in our training set. Then we'll drop rows from our training data that correspond to all indices less than or equal to our cutoff index that are also found in the set of exact duplicates flagged by `Datalab`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "df1fa776-a89d-454f-a42d-cfd8b73f2f1b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define training index cutoff and find the exact duplicate indices to reference\n",
+    "train_idx_cutoff = len(preprocessed_train_data) - 1\n",
+    "exact_duplicates_indices = exact_duplicates.index\n",
+    "\n",
+    "# Filter the indices to drop by which indices in exact duplicates are <= to the index cutoff\n",
+    "indices_of_duplicates_to_drop = [idx for idx in exact_duplicates_indices if idx <= train_idx_cutoff]"
```

Comment:
@mturk24 Are you sure this is correct?  We only want to drop the data points that are:
- in the training dataset
- exactly duplicated with some data point in the test set.

I don't see the test set appearing in any of the code here. It seems like you've just dropped all the training data points that are exact duplicates, even if theyre duplicated with other training data points only?  We don't want that

Reply:
You're correct, I am adjusting the code now to only drop the data points that are:

- in the training dataset
- exactly duplicated with some data point in the test set.

When rerunning the code and adjusting the logic, I actually find that there are NO data points that are **exact** duplicates between the training and test sets.

However, there are data points where `is_near_duplicate_issue` = `True` and `near_duplicate_score` is quite low as you can see in the screenshot below (715 is the cutoff index between training and test sets):

![Screenshot 2024-04-16 at 11 05 30 AM](https://github.com/cleanlab/cleanlab/assets/8604219/d9d07e51-d500-4329-a907-73d30a081a75)

Therefore, the behavior in the tutorial will be to drop no rows in the training set in this section since the only EXACT duplicates are intra-training set and not between training and test sets.

Reply:
see slack, your proposal is not what we want

Reply:
Addressed this in latest commit